services/venice-server/src/main/java/com/linkedin/venice/listener/response/stats/ComputeResponseStats.java

```diff
@@ -0,0 +1,110 @@
+package com.linkedin.venice.listener.response.stats;
+
+import static com.linkedin.venice.listener.response.stats.ResponseStatsUtil.consumeDoubleAndBooleanIfAbove;
+import static com.linkedin.venice.listener.response.stats.ResponseStatsUtil.consumeIntIfAbove;
+
+import com.linkedin.venice.stats.ServerHttpRequestStats;
+
+
+public class ComputeResponseStats extends MultiKeyResponseStats {
+  private double readComputeLatency = 0;
+  private double readComputeDeserializationLatency = 0;
+  private double readComputeSerializationLatency = 0;
+  private int readComputeOutputSize = 0;
+  private int totalValueSize = 0;
+  private int dotProductCount = 0;
+  private int cosineSimilarityCount = 0;
+  private int hadamardProductCount = 0;
+  private int countOperatorCount = 0;
+
+  @Override
+  public void addValueSize(int size) {
+    this.totalValueSize += size;
+  }
+
+  @Override
+  public void addReadComputeLatency(double latency) {
+    this.readComputeLatency += latency;
+  }
+
+  @Override
+  public void addReadComputeDeserializationLatency(double latency) {
+    this.readComputeDeserializationLatency += latency;
```

Comment:
This may not be `ThreadSafe` for parallel processing of compute requests, right?